### PR TITLE
Require C++14, enforce ALPS_CXX_STD, and remove Boost.Math

### PR DESCRIPTION
**Pull request by egull**
## Summary

Makes C++14 the minimum and default standard, makes `ALPS_CXX_STD` actually control the standard, and removes the Boost.Math dependency.

### Why
- `ALPS_CXX_STD` only added the `cxx_std_XX` compile feature, which is a *minimum*. CMake then passed no `-std` flag, so compilers with a newer default (AppleClang, GCC ≥ 11) silently built ALPSCore as C++17 even with the default `c++11`.
- Boost.Math requires C++14 since Boost 1.82, so C++11 builds break with current Boost. (The tests are compiled as C++17, as required by the bundled googletest 1.18 since #668.)

### Changes
1. **Remove Boost.Math** (`3f378ed1`)
   - alea: Hotelling T² test uses a small standalone F distribution (`alps/alea/fisher_f.hpp`: regularized incomplete beta via continued fraction, log B(a,b) with Stirling correction, overflow-safe in the parameters). Measured against Boost.Math over 48,000 random cases: relative error ~1e-13 for dof ≤ 1e3, ~1e-10 for dof ≤ 1e6, ~2e-7 for dof ≤ 1e9. New unit test `alea/test/fisher_f.cpp` checks reference values with matching tolerances, extreme parameters and edge cases.
   - `t2_result::dist()` now returns `alps::alea::fisher_f_distribution` (`degrees_of_freedom1/2()`, `cdf()`, `ccdf()`); `quantile()` is no longer available (it was only used for diagnostic output in `alea/test/model.cpp`).
   - utilities: `std::asinh/acosh/atanh`; tests use `std::isinf/isnan`.
   - `binning_analysis.hpp`: include `<iomanip>` (was only pulled in through Boost.Math).
2. **Enforce the requested standard** (`3513bd0b`): set `CMAKE_CXX_STANDARD` from `ALPS_CXX_STD`. Downstream projects still only get `cxx_std_XX` as a minimum.
3. **Require C++14** (`0b9db94b`)
   - `ALPS_CXX_STD` defaults to `c++14`; `c++11` is rejected like `c++03`. Fixes the validation regex (e.g. `gnu++14` was accepted).
   - Simplifies the standard selection to a single `cxx_std_XX` derived from `ALPS_CXX_STD`.
   - Minimum compilers: GCC 5, Clang 3.4, Intel 17.
   - `ALPS_FORCE_CXX_STD` accepts C++14/17/20.
   - Travis c++11 jobs → c++17; gf tutorial → C++14; INSTALL.md / CONTRIBUTING.md updated (libraries C++14, tests C++17).

## Testing (macOS, AppleClang, CMake 4.2.1, MPI on, Boost 1.92, rebased on #668 and #670)
- Default build: library sources compile with `-std=c++14`, tests and googletest with `-std=c++17`; 136/136 tests pass.
- Enforced C++20 build: 136/136 tests pass; C++17 builds cleanly.
- Installed and built all tutorials against the install.
- `ALPS_CXX_STD=c++11`/`c++03`/`gnu++14` rejected at configure time.
- Control: `master` with forced `-std=c++11` and Boost 1.92 fails in Boost.Math headers.

### CODE REVIEW

File: alea/src/fisher_f.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+double fisher_f_distribution::cdf(double f) const
+{
+    if (!(d1_ > 0 && d2_ > 0) || std::isnan(f))
+        return std::numeric_limits<double>::quiet_NaN();
+    if (f <= 0)
+        return 0;
+    if (std::isinf(f))
+        return 1;
+    double denom = d1_ * f + d2_;
+    return beta_inc(d1_ / 2, d2_ / 2, d1_ * f / denom, d2_ / denom);
```
[the remark this answers is not in the document]
Comment (egull) [reply alea/src/fisher_f.cpp]: Fixed in 6024579e. The beta arguments and their logs are now computed from q = f·d1/d2 with `log1p`, so `f·d1` never forms, and the continued-fraction coefficients are written as ratios. Note that for d1 = d2 = 1e308 the distribution is concentrated at 1, so the true P(F ≤ 2) is ≈ 1 (not 2/3); the new `no_overflow` test checks cdf(2) = 1, ccdf(2) = 0 and cdf(0.5) = 0 for d = 1e20, 1e200, 1e308, plus a case where f·d1 overflows. While there, log B(a,b) now uses the Stirling correction to avoid lgamma cancellation (worst relative error vs Boost.Math at dof ~1e9 drops from ~1e-5 to ~2e-7), and the continued fraction returns NaN instead of an unconverged value.

File: alea/test/fisher_f.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// Relative accuracy is better than 1e-12 for moderate degrees of freedom, degrading to
+// ~1e-9 for d2 ~ 1e6 (cancellation between large lgamma values).
+TEST(fisher_f, reference)
+{
+    for (const auto &r : reference) {
+        fisher_f_distribution dist(r[0], r[1]);
+        EXPECT_NEAR(dist.cdf(r[2]), r[3], 1e-8 * r[3])
+            << "d1=" << r[0] << " d2=" << r[1] << " f=" << r[2];
+        EXPECT_NEAR(dist.ccdf(r[2]), r[4], 1e-8 * r[4])
+            << "d1=" << r[0] << " d2=" << r[1] << " f=" << r[2];
```
[the remark this answers is not in the document]
Comment (egull) [reply alea/test/fisher_f.cpp]: Fixed in 6024579e. I measured accuracy against Boost.Math over 48,000 random (d1, d2, f) samples spanning dof 0.5 to 1e9 and quantiles 1e-12 to 1 - 1e-12: worst relative error ~1e-13 for dof ≤ 1e3, ~1e-10 for dof ≤ 1e6, ~2e-7 for dof ≤ 1e9. That contract is now documented in `fisher_f.hpp`, and the test uses matching per-row tolerances (1e-12 / 1e-9 / 1e-6 by the largest dof) with additional reference rows for large dof and extreme f.

File: alea/src/fisher_f.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    return front == 0 ? 1 : 1 - front * beta_cf(b, a, y) / b;
+}
+
+// P(F <= f) or P(F > f) for f > 0, using the ratio q = f d1/d2 so that
+// nothing overflows: x = q/(1 + q), y = 1/(1 + q).
+double f_tail(double d1, double d2, double f, bool upper)
+{
+    double q = d1 / d2 * f;
+    if (std::isinf(q))
+        return upper ? 0 : 1;
```
[the remark this answers is not in the document]
Comment (egull) [reply alea/src/fisher_f.cpp]: Fixed in 63c16edf. If `d1/d2·f` over- or underflows, log q = log d1 − log d2 + log f is used instead, the tail cut-offs are based on log x / log y rather than x or y underflowing to zero, and results are clamped to [0, 1]. For d1 = 1, d2 = 1e-309, f = 0.1 this now gives cdf = 0, ccdf = 1, matching Boost.Math. Regression cases are in the `no_overflow` test. For degrees of freedom far below 1, only an absolute accuracy of ~1e-13 holds; this is documented in `fisher_f.hpp`.

File: common/cmake/ALPSCompilerVersionCheck.cmake [hunk 4]
```diff
@@ -79,53 +79,19 @@ if (ALPS_CXX_STD STREQUAL "custom")
...
-      else()
-        set(ALPS_CMAKE_MINIMUM_VERSION 3.8)
-        set(ALPS_CXX_FEATURES "cxx_std_20")
-      endif()
-      message(STATUS "ALPSCore will use C++20")
-    endif()
+  # Does CMake know how to use C++XX for this compiler?
+  if (CMAKE_CXX_COMPILE_FEATURES)
+    # Downstream projects need at least this standard
+    set(ALPS_CXX_FEATURES "cxx_std_${alps_cxx_std_num_}")
```
[the remark this answers is not in the document]
Comment (egull) [reply common/cmake/ALPSCompilerVersionCheck.cmake]: Resolved by #668 (now merged, this PR is rebased on it): the CMake minimum is now 3.16 for ALPSCore and for projects that use it (`ALPSCore_CMAKE_MINIMUM_VERSION`). `cxx_std_20` has been known to CMake since 3.12, so it is always available at the supported minimum.

File: common/cmake/ALPSCoreConfig.cmake.in [hunk 5]
```diff
@@ -21,12 +21,10 @@
...
-      if (ALPS_FORCE_CXX_STD MATCHES "^[Cc][+][+]11$")
-        set(cxx_features "cxx_auto_type;cxx_constexpr")
-        message(STATUS "ALPSCore will request C++11 features")
-      elseif (ALPS_FORCE_CXX_STD MATCHES "^[Cc][+][+]14$")
-        set(cxx_features "cxx_decltype_auto")
-        message(STATUS "ALPSCore will request C++14 features")
+      if (ALPS_FORCE_CXX_STD MATCHES "^[Cc][+][+](14|17|20)$")
+        string(REGEX REPLACE "^[Cc][+][+]" "" std_num_ ${ALPS_FORCE_CXX_STD})
+        set(cxx_features "cxx_std_${std_num_}")
+        message(STATUS "ALPSCore will request C++${std_num_} features")
```
[the remark this answers is not in the document]
Comment (egull) [reply common/cmake/ALPSCoreConfig.cmake.in]: Resolved by #668 (now merged, this PR is rebased on it): the CMake minimum is now 3.16 for ALPSCore and for projects that use it (`ALPSCore_CMAKE_MINIMUM_VERSION`). `cxx_std_20` has been known to CMake since 3.12, so it is always available at the supported minimum.

File: alea/src/fisher_f.cpp [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+// underflow to zero while their logarithms are still finite).
+double beta_inc(double a, double b, double x, double y, double log_x, double log_y)
+{
+    if (std::isinf(log_x)) return 0;
+    if (std::isinf(log_y)) return 1;
+    double front = std::exp(a * log_x + b * log_y - log_beta(a, b));
+    double result = x < (a + 1) / (a + b + 2)
+        ? (front == 0 ? 0 : front * beta_cf(a, b, x) / a)
+        : (front == 0 ? 1 : 1 - front * beta_cf(b, a, y) / b);
+    return std::fmin(std::fmax(result, 0.), 1.);
```
[the remark this answers is not in the document]
Comment (egull) [reply alea/src/fisher_f.cpp]: Good catch, fixed in a4a068ae: a NaN result is now returned before clamping. The new test `no_convergence_is_nan` (d1 = d2 = 1e14 at f = 1, where the continued fraction does not converge) fails without the fix and passes with it.
